shotover-proxy/src/server.rs

```diff
@@ -253,6 +253,24 @@ impl<C: Codec + 'static> TcpCodecListener<C> {
         }
     }
 
+    pub async fn shutdown(&mut self) {
+        match self
```

Comment:
we probably want this to timeout if the whole chain is taking too long to shutdown (configurable)

Reply:
Won't it just be SIGKILL'd if it takes too long

Reply:
only if someone notices it's stuck... most workflows people will just see a stuck autoscaling group in AWS or a stuck stateful set in kubernetes or something

Reply:
Kubernetes will kill any processes that don't shut down in 30 seconds after a SIGTERM. Not sure about AWS but I would imagine it would do something similar if a process was refusing to shutdown.

Reply:
ok happy to leave this up to an external scheduler to do